Backend/auction-app/src/main/java/com/example/auctionapp/repository/AbstractJpaDao.java

```diff
@@ -0,0 +1,44 @@
+package com.example.auctionapp.repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+public abstract class AbstractJpaDao<T> {
+
+    private Class<T> resourceClass;
+
+    @PersistenceContext
+    EntityManager entityManager;
+
+    public void setResourceClass( Class< T > classToSet ) {
+        this.resourceClass = classToSet;
+    }
+
+    public T findById(Long id){
+        return entityManager.find(resourceClass, id);
+    }
+
+    public List<T> findAll(){
+        return entityManager.createQuery("from " + resourceClass.getName())
+                .getResultList();
+    }
```

Comment:
We should always avoid hardcoding anything. If there is a way to write something in code rather than using hardcoded strings we should do it. For example, creating the query in string is a bad idea for future changes and it is prone to mistakes.
The way to do this in code is to use the `entityManager's criteriaBuilder API  `. 

Investigate about this API as this will be the one you will use the most for building queries.